ocaml/xapi/xapi_pusb.ml

```diff
@@ -88,3 +88,60 @@ let scan_thread ~__context =
   in
   start_thread f;
   scan ~__context
+
+let get_sm_usb_path ~__context vdi =
+  try
+    let vdi_sc = Db.VDI.get_sm_config ~__context ~self:vdi in
+    List.assoc Xapi_globs.usb_path vdi_sc
+  with _ -> ""
+
+let set_passthrough_enabled ~__context ~self ~value =
+  Mutex.execute mutex (fun () ->
+    match value with
+     | true ->
+       (* Remove the vdi records which 'usb_path' in sm-config has the
+          same value with the field 'path' in PUSB. *)
+       let udev_srs = Db.SR.get_refs_where ~__context ~expr:(Eq(Field "type", Literal "udev")) in
+       List.iter (fun sr ->
+         Db.VDI.get_refs_where ~__context ~expr:(Eq(Field "SR", Literal (Ref.string_of sr))) |>
+           List.iter (fun rf ->
+             try
+               if (get_sm_usb_path ~__context rf) = (Db.PUSB.get_path ~__context ~self) then begin
```

Comment:
Can the call to `(Db.PUSB.get_path ~__context ~self)` inside the iter-loop be avoided by doing it once outside and binding the result with a let?

Reply:
updated.